### ISSUE TITLE
p4 annotate invalid option -u

### ISSUE THREAD

**Issue by unional** [stef-levesque/vscode-perforce#75]
Currently `vscode-perforce` runs: `p4 annotate -qu <file>`

Results in `invalid option -u`.

Use the latest version of p4v (2017.1) which comes with p4 (2016.2) and it does not have this flag.

Where do you get the p4 2017.1 version?

Is there a way to exclude the `-u` so it can work with this version of p4?

**Comment by unional** [comment]
FYI:
```sh
p4 help annotate

    annotate -- Print file lines and their revisions

    p4 annotate [-aciIqt -d<flags>] file[revRange] ...

        Prints all lines of the specified files, indicating the revision that
        introduced each line into the file.

        If the file argument includes a revision, then only revisions up to
        the specified revision are displayed.  If the file argument has a
        revision range, only revisions within that range are displayed. For
        details about specifying revisions, see 'p4 help revisions'.

        The -a flag includes both deleted files and lines no longer present
        at the head revision. In the latter case, both the starting and ending
        revision for each line is displayed.

        The -c flag directs the annotate command to output changelist numbers
        rather than revision numbers for each line.

        The -d<flags> change the way whitespace and/or line endings are
        treated: -db (ignore whitespace changes), -dw (ignore whitespace),
        -dl (ignore line endings).

        The -i flag follows branches.  If a file was created by branching,
        'p4 annotate' includes the revisions of the source file up to the
        branch point, just as 'p4 filelog -i' does.  If a file has history
        prior to being created by branching (such as a file that was branched
        on top of a deleted file), -i ignores those prior revisions and
        follows the source.  -i implies -c.

        The -I flag follows all integrations into the file.  If a line was
        introduced into the file by a merge, the source of the merge is
        displayed as the changelist that introduced the line. If the source
        itself was the result of an integration, that source is used instead,
        and so on.  -I implies -c and may not be combined with -i.

        The -q flag suppresses the one-line header that is displayed by
        default for each file.

        The -t flag forces 'p4 annotate' to display binary files.
```

**Comment by stef-levesque** [comment]
This ˋ-uˋ flag exists since release 15.2
https://www.perforce.com/perforce/r15.2/manuals/cmdref/p4_annotate.html

> Display the name of the user who modified the change and the date when the modification occurred.

Are you sure your version of p4.exe is >= 15.2 ?

**Comment by unional** [comment]
I do think it is 2016.2, I just install it again.

However, the server is 2014.2  I don't know if this would impact the p4.exe flags.

**Comment by nomaed** [comment]
Same issue.
Server is 2015.1 but client was just downloaded is it's on latest version: 2017.1 (Mac)
However:
```
p4 help annotate

    annotate -- Print file lines and their revisions

    p4 annotate [-aciIqt -d<flags>] file[revRange] ...
```

There is no -u flag for me.
It is possible that it's determined by the server version.

At any rate, annotate doesn't work.

**Comment by stef-levesque** [comment]
Should be fixed in version 2.2.0. 
Please update and try setting `perforce.annotate.user` to `false`
